src/Lykke.Service.History/Workflow/Projections/TransactionHashProjection.cs

```diff
@@ -26,11 +28,8 @@ public TransactionHashProjection(IHistoryRecordsRepository historyRecordsReposit
         public async Task<CommandHandlingResult> Handle(CashinCompletedEvent @event)
         {
             if (!await _historyRecordsRepository.UpdateBlockchainHashAsync(@event.OperationId, @event.TxHash))
-                _logger.Warning($"Transaction hash was not set, bitcoin cashin", context: new
-                {
-                    id = @event.OperationId,
-                    hash = @event.TxHash
-                });
+                throw new InvalidOperationException($"Transaction hash was not set. " +
```

Comment:
Why throwing exception? It seems like log.Warning + Retry should be the right choice here. The same applies to the code below.